Pull-Request: integration with clang and iedit.

yangyingchao opened the pull request:
Key points:
1. Integrated iedit to rename variables/functions.
2. Integrated clang to format codes for c/c++.
3. Reordered header files for C/C++.

Thanks,

chrisbarrett commented:
Thanks for the PR. Those external packages are all available on MELPA, so you shouldn't add the sources--just add them to the ELPA package includes [HERE]().

HERE: https://github.com/chrisbarrett/emacs-refactor/blob/master/emr.el#L8

chrisbarrett commented:
I'll be pleased to merge once you update your PR to remove those third-party libs. :)

yangyingchao commented:
It's done.

Thanks

chrisbarrett commented on emr.el:
```diff
@@ -340,6 +341,16 @@ Return a popup item for the refactoring menu if so."
   (eval-after-load 'js2-refactor  '(require 'emr-js))
   (eval-after-load 'ruby-refactor '(require 'emr-ruby)))
 
+(defun emr:try-require (feature)
```
Looks like this function isn't used.

chrisbarrett commented:
Sorry for letting this drop off my radar. :)

yangyingchao commented:
Dead codes removed.

BTW, I have some more aggressive changes, you can find them here:

https://github.com/yangyingchao/tubo-env/tree/master/.emacs.d/site-lisp/emr/

Main changes:
1. Moved common functions which work for all modes derived from cc-mode into emr-cc.el, for example, c-mode, c++-mode, objc-mode.
2. Added emr-cpp-try-catch for c++-mode. This can be used to surround a segment of code with try-catch clause.
3. Added emr-org-quote for org-mode, which can be used to quote a region of code quickly.
4. Added support to customize clang-format with user defined alist.


I didn't merge above changes into this repository since I'm not sure how much it will help others.

yangyingchao commented:
Dead codes removed.

BTW, I have some more aggressive changes, you can find them here:

https://github.com/yangyingchao/tubo-env/tree/master/.emacs.d/site-lisp/emr/

Main changes:
1. Moved common functions which work for all modes derived from cc-mode into emr-cc.el, for example, c-mode, c++-mode, objc-mode.
2. Added emr-cpp-try-catch for c++-mode. This can be used to surround a segment of code with try-catch clause.
3. Added emr-org-quote for org-mode, which can be used to quote a region of code quickly.
4. Added support to customize clang-format with user defined alist.


I didn't merge above changes into this repository since I'm not sure how much it will help others.
